### ISSUE TITLE
Disable Markdown images

### ISSUE THREAD

**Issue by M1dgard** [mattermost/mattermost-server#8462]
#### Summary
Inlining images with `![alt text](https://example.org/image.jpg)` should be (possible to) disable(d).

#### Steps to reproduce
Post a message with `![](http://upload.wikimedia.org/wikipedia/commons/b/ba/Kammergericht%2C_Berlin-Sch%C3%B6neberg%2C_Treppenhalle_%281%29%2C_160809%2C_ako.jpg)` (note: http, no https)

#### Expected behavior
Mattermost does not allow the image syntax to be translated to an HTML img element.

#### Observed behavior (that appears unintentional) 
If your browser allows mixed content, the image does show.
If it doesn't allow mixed content, it may show a mixed content warning.

#### Rationale
* They can leak the URL of private Mattermost instances to third parties via the Referer.
* I want a 3rd-party free experience on my Mattermost instance.
* They make the browser issue a mixed content warning if someone posts an image with a non-HTTPS URL.
* They can leak users' IP addresses to other users (if they craft a URL to their own server).
* They facilitate CSRF attacks (but that's really the fault of the victim site of course).
* They are annoying.
* They can incur data costs for mobile users.

#### Possible fixes
Add a configuration option for admins to disable the Markdown image syntax, or disable it altogether.

---

I don't know if you will consider this more a feature request than a bug, but that User Voice site doesn't work for me anyway so I figured I'd post it here. Feel free to migrate.

**Comment by SmartHoneybee** [comment]
Hi @M1dgard,

an [image proxy][1] might help you to address some of those points.

Regards,
Bee

[1]: https://docs.mattermost.com/administration/image-proxy.html

**Comment by M1dgard** [comment]
Solved by using an image proxy:

:white_check_mark: They can leak the URL of private Mattermost instances to third parties via the Referer.
:white_check_mark: I want a 3rd-party free experience on my Mattermost instance.
:white_check_mark: They make the browser issue a mixed content warning if someone posts an image with a non-HTTPS URL.
:white_check_mark: They can leak users' IP addresses to other users (if they craft a URL to their own server).
:white_check_mark: They facilitate CSRF attacks (but that's really the fault of the victim site of course).
:negative_squared_cross_mark: They are annoying.
:negative_squared_cross_mark: They can incur data costs for mobile users.

What I did was set the image proxy URL to `https://null.localhost`, now the last two points are solved as well. Thanks.
